### ISSUE TITLE
Support for token.validator.class implementor to read application properties

### ISSUE THREAD

**Issue by kmarx** [drajer-health/eCRNow#559]
We have our own `token.validator.class`. We'd like to have it read some new custom application properties (such as an auth endpoint). 

You can probably ignore the rest of the description here and simply see the **"Update"** comment below.

<<<ignore
But we can't seem to get any kind of `@Value("${...}")` or `@ComponentScan` annotations to make the properties visible to the class.

We _can_ get it to work by adding our package name to the basePackages component scan in `SrpingConfiguration`. For this we tried to be as generic as possible by adding `"com.vendor"` to the list of packages and creating our validator class as `com.vendor.ourcompany.OurTokenValidator`. Even then, the `@Value()` annotations don't really work, so we also had to make that class implement `EnvironmentAware` and add a `setEnvironment()` override to grab the desired properties and stash in static variables.

Plus, we then see the `doFilterInternal()` method being invoked twice for each request: Once for the class from the `token.validator.class` property injection, and once for the `$$EnhancerBySpringCGLIB$$` class dynamically created by the component scan. We had to hack things to filter for just the former.

We're hoping there's a way to support this more formally and cleanly in the Drajer project? If not, at least perhaps add the empty "com.vendor" package to the SpringConfiguration list and let vendors hack through the making it work as we did.
<<</ignore

**Comment by kmarx** [comment]
**Update:**
Another team member found this solution that requires a fairly simple change to `WebSecurityConfig.java` where we can autowire the token filter class instance:

```
+  @Autowired
+  private ApplicationContext context;
+
   @Value("${token.validator.class}")
   private String tokenFilterClassName;
 
@@ -34,10 +39,11 @@ public class WebSecurityConfig extends WebSecurityConfigurerAdapter {
       logger.info("Token Filter class Name is not empty");
       Class<?> classInstance = Class.forName(tokenFilterClassName);
       logger.info(classInstance.getDeclaredMethods()[0].getName());
+      Filter customFilter = (Filter) context.getAutowireCapableBeanFactory().autowire(classInstance,1,true);
       http.csrf()
           .disable()
           .addFilterAfter(
-              (Filter) classInstance.newInstance(), UsernamePasswordAuthenticationFilter.class)
+              customFilter, UsernamePasswordAuthenticationFilter.class)
           .authorizeRequests()
           .antMatchers("/api/**")
           .permitAll()
```

Seems to work perfectly, allowing visibility of injected properties and is backwards compatible with other `token.validator.class` implementors.

**Comment by kmarx** [comment]
One additional thought that the team member had was that you might want to support auto wiring more than one custom class. The property could be a _list_ of classes that you iterate through as above.

**Comment by nbashyam** [comment]
Will add to 3.1.2

**Comment by nbashyam** [comment]
Fixed in 3.1.2 , added an application.properties property for adding the validator class.
